app/geant-exporter/geant-exporter.cc

```diff
@@ -234,13 +161,71 @@ int main(int argc, char** argv)
             table_parser.add_physics_table(process, particle);
         }
     }
-
     std::cout << root_output_filename << " done!" << std::endl;
+}
 
-    //---------------------------- Job termination --------------------------//
+//---------------------------------------------------------------------------//
+/*!
+ * MAIN
+ */
+int main(int argc, char** argv)
+{
+    //-----------------------------------------------------------------------//
     /*!
-     * User actions, physics list, and detector construction are owned and
-     * deleted by the run manager
+     * Minimal help
      */
-    delete run_manager;
+    if (argc != 2)
+    {
+        G4cout << "Usage: ./geant4-exporter geometry.gdml" << G4endl;
+        return 1;
+    }
+
+    G4RunManager run_manager;
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Fetching the GDML input file and initializing the geometry
+     */
+    std::string gdml_input = argv[argc - 1];
+    auto detector = std::make_unique<DetectorConstruction>(gdml_input);
+    run_manager.SetUserInitialization(detector.release());
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Loading the physics list
+     */
+    auto physics_constructor = std::make_unique<std::vector<G4String>>();
+    physics_constructor->push_back("G4EmStandardPhysics");
+
+    auto physics_list = std::make_unique<G4GenericPhysicsList>(
+        physics_constructor.release());
+
+    // For the full Physics List:
+    // Uncomment #include <FTFP_BERT.hh> and use the following pointer instead
+    // auto physics_list = std::make_unique<FTFP_BERT>();
+
+    run_manager.SetUserInitialization(physics_list.release());
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Starting a minimal run to produce the physics tables
+     */
+    auto action_initialization = std::make_unique<ActionInitialization>();
+    run_manager.SetUserInitialization(action_initialization.release());
+    
+    G4UImanager* ui_manager = G4UImanager::GetUIpointer();
+    ui_manager->ApplyCommand("/run/initialize");
+    run_manager.BeamOn(1);
+
+    //-----------------------------------------------------------------------//
+    /*!
+     * Storing both particle information and physics tables
+     */
+    std::string root_output_filename = "geant-exporter-data.root";
+
+    G4ParticleTable::G4PTblDicIterator* particle_iterator
+        = G4ParticleTable::GetParticleTable()->GetIterator();
+
+    store_particles(root_output_filename, particle_iterator);
+    store_physics_tables(root_output_filename, particle_iterator);
```

Comment:
Much easier to read! Thanks! One minor question is whether `particle_iterator` is something that you own or not? I think it should be passed around as a reference regardless. The semantics of the iterator aren't clear when raw pointers are involved. The fact that it's mutable also makes it unclear whether any sort of "reset" has to be called between your `store_` functions.

Reply:
I should not own it, since as far as I know, all is owned by the Run Manager. An `iterator->reset()` is called inside each `store_` function.